leaderboard/service/get_around_me_test.go

```diff
@@ -0,0 +1,335 @@
+
+			_, err := svc.GetAroundMe(context.Background(), leaderboard, pageSize, member, order, getLastIfNotFound)
+			Expect(err).To(Equal(service.NewGeneralError("get around me", "database error")))
+		})
+
+		It("Should return error if TotalMembers return in error", func() {
+			mock.EXPECT().GetRank(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(member), gomock.Eq(order)).Return(-1, database.NewMemberNotFoundError(leaderboard, member))
+			mock.EXPECT().GetTotalMembers(gomock.Any(), gomock.Eq(leaderboard)).Return(-1, fmt.Errorf("database error"))
+
+			_, err := svc.GetAroundMe(context.Background(), leaderboard, pageSize, member, order, getLastIfNotFound)
+			Expect(err).To(Equal(service.NewGeneralError("get around me", "database error")))
+		})
+
+		It("Should return error if GetOrderedMembers return in error", func() {
+			start, stop := calculateIndexes(totalMembers, pageSize, totalMembers)
+
+			mock.EXPECT().GetRank(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(member), gomock.Eq(order)).Return(-1, database.NewMemberNotFoundError(leaderboard, member))
+			mock.EXPECT().GetTotalMembers(gomock.Any(), gomock.Eq(leaderboard)).Return(totalMembers, nil)
+			mock.EXPECT().GetTotalMembers(gomock.Any(), gomock.Eq(leaderboard)).Return(totalMembers, nil)
+			mock.EXPECT().GetOrderedMembers(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(start), gomock.Eq(stop), gomock.Eq(order)).Return(nil, fmt.Errorf("database error"))
+
+			_, err := svc.GetAroundMe(context.Background(), leaderboard, pageSize, member, order, getLastIfNotFound)
+			Expect(err).To(Equal(service.NewGeneralError("get around me", "database error")))
+		})
+
+		It("Should ask for all members if totalMembers is less than pageSize", func() {
+			var totalMembers int = 2
+			start := 0
+			stop := 2
+
+			mock.EXPECT().GetRank(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(member), gomock.Eq(order)).Return(-1, database.NewMemberNotFoundError(leaderboard, member))
+			mock.EXPECT().GetTotalMembers(gomock.Any(), gomock.Eq(leaderboard)).Return(totalMembers, nil)
+			mock.EXPECT().GetTotalMembers(gomock.Any(), gomock.Eq(leaderboard)).Return(totalMembers, nil)
+
+			//this is the assertation relevant to this test
+			mock.EXPECT().GetOrderedMembers(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(start), gomock.Eq(stop), gomock.Eq(order)).Return(nil, fmt.Errorf("database error"))
+
+			_, err := svc.GetAroundMe(context.Background(), leaderboard, pageSize, member, order, getLastIfNotFound)
+			Expect(err).To(Equal(service.NewGeneralError("get around me", "database error")))
+		})
+	})
+})
+
+func calculateIndexes(totalMembers, pageSize, rank int) (int, int) {
+	start := rank - (pageSize / 2)
+	if start < 0 {
+		start = 0
+	}
+
+	stop := start + pageSize - 1
+	if stop > totalMembers {
+		stop = totalMembers
+		start = stop - pageSize + 1
+		if start < 0 {
+			start = 0
+		}
+	}
+
+	return start, stop
+}
```

Comment:
You are implementing this twice. Here and on indexes.go file. If you extract this code to a public function (might not be in the services module though), you can reuse it in both places.

Reply:
According to the last PR, I thought that absolute values are better than use the same function to calculate the indexes because an error on this function no test will get them, and absolutely values a future change will get it

Reply:
So in my changes I remove this function

Reply:
![nice](https://i.pinimg.com/originals/99/5a/d5/995ad537dcb285d682b39c0777e9b640.gif)